File: src/jane/handler/testclient/TestBeanHandler.java
package jane.handler.testclient;

import jane.bean.TestBean;
import jane.core.BeanHandler;
import jane.core.Log;
import jane.core.NetManager;
import jane.test.TestRc4Filter;
import org.apache.mina.core.session.IoSession;

public final class TestBeanHandler implements BeanHandler<TestBean>
{
	@Override
	public TestBean beanStub()
	{
		return TestBean.BEAN_STUB;
	}

	/*\
	|*| int TEST_CONST1 = 5; // 测试类静态常量
	|*| String TEST_CONST2 = "test_const2";
	|*| int value1; // 字段的注释
	|*| long value2;
	\*/

	@Override
	public void onProcess(NetManager manager, IoSession session, TestBean arg)
	{
		Log.debug("{}: arg={}", getClass().getName(), arg);

		TestRc4Filter filter = new TestRc4Filter();
		filter.setInputKey(new byte[] { 1, 2, 3 }, 3);
		filter.setOutputKey(new byte[] { 1, 2, 3 }, 3);
		session.getFilterChain().addFirst("enc", filter);

		manager.send(session, arg);
	}
}

File: port/csharp/jane/Handler/TestBeanHandler.cs
using Jane.Bean;

namespace Jane.Handler
{
	public class TestBeanHandler
	{
		/*\
		|*| int TEST_CONST1 = 5; // 测试类静态常量;
		|*| string TEST_CONST2 = "test_const2";
		|*| int value1; // 字段的注释;
		|*| long value2;
		\*/

		public static void OnProcess(NetManager.NetSession session, IBean _arg_)
		{
			TestBean arg = (TestBean)_arg_;
			System.Console.WriteLine("{0}.onProcess: arg={1}", arg.GetType().Name, arg);

			RC4Filter filter = new RC4Filter();
			filter.SetInputKey(new byte[] { 1, 2, 3 }, 3);
			filter.SetOutputKey(new byte[] { 1, 2, 3 }, 3);
			((TestClient)session.manager).SetFilter(filter);

			session.Send(arg);
		}
	}
}
